File: languages/go/entity/availability/loader_args.go
package availability

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// LoaderArgs represents the arguments for a data loader.
type LoaderArgs struct {
	// Id is a unique identifier associated with the loader.
	Id string
}

func (e LoaderArgs) ToProto() *proto.LoaderArgs {
	return &proto.LoaderArgs{
		Id: e.Id,
	}
}

File: languages/js/src/entity/availability/loader_args.ts
import * as proto from "../../bridge/proto/bloock_availability_entities";

/**
 * Represents the arguments for a data loader.
 */
export class LoaderArgs {
  /**
   * Is a unique identifier associated with the loader.
   */
  id: string;

  /**
   * Constructs a LoaderArgs object with the specified parameters.
   * @param id
   */
  constructor(id: string) {
    this.id = id;
  }

  public toProto(): proto.LoaderArgs {
    return proto.LoaderArgs.fromPartial({ id: this.id });
  }
}
